src/vm_pages.rs

```diff
@@ -1805,6 +1811,80 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         Ok(())
     }
 
+    // Promotes a page range, creating one page of the next larger page size.
+    pub fn promote_page(&self, page_addr: GuestPageAddr, page_size: PageSize) -> Result<()> {
+        let end = page_addr
+            .checked_add_pages_with_size(1, page_size)
+            .ok_or(Error::AddressOverflow)?;
+        let regions = self.inner.regions.read();
+        if !(regions.contains(page_addr, end, |r| r == VmRegionType::Confidential)
+            || regions.contains(page_addr, end, |r| r == VmRegionType::ConfidentialRemovable)
+            || regions.contains(page_addr, end, |r| r == VmRegionType::Shared)
+            || regions.contains(page_addr, end, |r| r == VmRegionType::SharedRemovable))
```

Comment:
nit: check the possible region types in the closure rather than calling `contains()` 4 times. same goes for `demote_page()`

Reply:
That's actually on purpose that I'm calling 4 times the `contains()` function, because I want to make sure the whole region contains the same type. If I move to a single check with the closure handling all the types at once, that means we're okay promoting a page on a range that might be of different types.

Reply:
`contains()` won't span regions; the range must fall entirely in one region.

Reply:
Ah good point :)